crates/common/clock/src/lib.rs

```diff
@@ -16,3 +20,58 @@ impl Clock for WallClock {
         OffsetDateTime::now_utc()
     }
 }
+
+#[automock]
+pub trait MonotonicClock: Send + Sync + 'static {
+    fn now(&self) -> TedgeInstant;
+}
+
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+/// A wrapper type to prevent accidental use of [Instant::elapsed]
+///
+/// Once a time is produced by [MonotonicClock], we don't want to use
+/// [Instant::elapsed] as that will compare to [Instant::now] and not
+/// [MontonicClock::now].
+pub struct TedgeInstant(Instant);
+
+pub struct SystemClock;
+
+impl MonotonicClock for SystemClock {
```

Comment:
I'm okay with that. However, have you considered [`tokio::time::pause`](https://docs.rs/tokio/latest/tokio/time/fn.pause.html) and  [`tokio::time::advance`](https://docs.rs/tokio/latest/tokio/time/fn.advance.html)?

Reply:
I'd forgotten about those, I'll change to using them since this introduces unnecessary complexity.

Reply:
Indeed, far simpler!